md/codeBlock/types.ts

```diff
@@ -1,6 +1,11 @@
 import { FencedCodeBlockDetails } from "./fenced.ts";
 import { IndentedCodeBlockDetails } from "./indented.ts";
 
+export type CodeBlockInfo = {
+  code: string;
+  lineNumber: number;
```

Comment:
After reviewing the rest of the PR, my recommendation is:

- Create a new file `string/location.ts`

  - `export type Location = { offset:number, line: number, column: number }`

  - `export location(input: string, offset: number): Location` or some similar name

Then use those here. This will work for you, will work as a generally useful util, and will work for my parsing work as well.

Reply:
Now using `string/location.ts` - just merged!